### PR TITLE
feature/44 add search function

### PR DESCRIPTION
**Pull request by NDMyroshnychenko**
Implement search client functionality

### CODE REVIEW

File: slidez-fe-core/src/pages/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -83,6 +84,25 @@ const Dashboard = () => {
...
+
+    const isNotEmptyPresentation = () => {
+        return filteredPresentation.length === 0 ? (
+            <span>No presentions are found by search term {searchField}.</span>
+        ) : (
+            ''
+        )
+    }
+
+    // @ts-ignore
```
Comment (Anatolii-Ivanov) [inline slidez-fe-core/src/pages/dashboard/Dashboard.tsx]: Don't use ts-ignore

File: slidez-fe-core/src/pages/dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -83,6 +84,25 @@ const Dashboard = () => {
         })
     }
 
+    const filteredPresentation = MOCK_DATA.filter((presentation) => {
+        return presentation.name
+            .toLowerCase()
+            .includes(searchField.toLowerCase())
+    })
```
Comment (Anatolii-Ivanov) [inline slidez-fe-core/src/pages/dashboard/Dashboard.tsx]: Please, use useEffect on searchField changes to filter presentations and store filtered presentations using useState()

File: slidez-fe-core/src/pages/dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -83,6 +86,32 @@ const Dashboard = () => {
...
     }
 
+    useEffect(() => {
+        setFilteredPresentations(
+            MOCK_DATA.filter((presentation) => {
+                return presentation.name
+                    .toLowerCase()
+                    .includes(searchField.toLowerCase())
+            })
+        )
```
Comment (Anatolii-Ivanov) [inline slidez-fe-core/src/pages/dashboard/Dashboard.tsx]: You can use it without return statement
MOCK_DATA.filter((presentation) => presentation.name
                    .toLowerCase()
                    .includes(searchField.toLowerCase())
            )

File: slidez-fe-core/src/pages/dashboard/Dashboard.tsx [hunk 4]
```diff
@@ -83,6 +86,32 @@ const Dashboard = () => {
...
+
+    const isNotEmptyPresentation = () => {
+        return filteredPresentations.length === 0 ? (
+            <span>
+                No presentations are found by search term {searchField}.
+            </span>
+        ) : (
+            ''
+        )
+    }
```
Comment (Anatolii-Ivanov) [inline slidez-fe-core/src/pages/dashboard/Dashboard.tsx]: The same

File: slidez-fe-core/src/pages/dashboard/Dashboard.tsx [hunk 5]
```diff
@@ -235,6 +266,7 @@ const Dashboard = () => {
                                         </div>
                                     </div>
                                 ))}
+                                {isNotEmptyPresentation()}
```
Comment (Anatolii-Ivanov) [inline slidez-fe-core/src/pages/dashboard/Dashboard.tsx]: Can you add styles to it?)
like 
HTML:
```<div class="no-found-presentaitons"><p>Some text here</p></div>```

CSS:
.no-found-presenation {
display:flex;
justify-content: center;
align-content: center;
width: 100%;
height: 100%; // Here can be some troubles
font-weight: 400;
font-size: 16px;
}

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by VL4DYSL4V — APPROVED** [review]

**Pull request merged**